src/processout/processout.ts

```diff
@@ -1253,8 +1294,13 @@ module ProcessOut {
                                     success: (data: any) => void,
                                     error: (err: Exception) => void): void {
 
+            // returns this.hppInitialURL only once during the first call from HPP, then returns the endpoint
+            const getEndpoint = (): string => {
+                return !this.threeDSInitiationRequested ? endpoint : (this.threeDSInitiationRequested = false, this.threeDSInitiationURL);
+            }
+
             if (!options) options = {};
-            var source = cardID;
+            let source: string = cardID;
```

Comment:
Do we need to explicitly say that it's string? Probably TypeScript should handle it on its own 🤔

Reply:
The source would be always a string, so I thought that adding the type was a nice idea
Anyway, my IDE is suggesting me to put string there

I am not sure about this, but what would you suggest? What would be the best option here?

Reply:
Ok, let's leave the type here, it was just nitpick 😄

Reply:
TypeScript should be smart enough to infere the type (take a look [here](https://www.typescriptlang.org/docs/handbook/type-inference.html)) but anyway we can be more explicit 👍